src/main/java/soot/FastHierarchy.java

```diff
@@ -763,10 +770,21 @@ private SootMethod resolveMethod(
       Set<SootClass> ignoreList) {
     SootClass concreteType = baseType;
 
+    
+    SootMethod candidate = null;
+    SmallNumberedMap<SootMethod> vtbl = typeToVtbl.get(baseType.getType());
+    if(vtbl == null) {
+    	typeToVtbl.put(baseType.getType(), vtbl = new SmallNumberedMap<SootMethod>());
+    }
+    if(methodName != null) {
+    	candidate = vtbl.get(methodName.getNumberedSubSignature());
+    	if(candidate != null) {
+    		return candidate;
+    	}
+    }
+    
     // When there is no proper dispatch found, we simply return null to let
     // the caller decide what to do
-    SootMethod candidate = null;
-
     while (concreteType != null && !ignoreList.contains(concreteType)) {
       ignoreList.add(concreteType);
```

Comment:
Dont forgett to add the resolved method to the vtable if one was found (at the end of the method)